subprojects/core/src/main/java/org/gradle/api/internal/project/ProjectDirectoryProvider.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.internal.project;
+
+import java.io.File;
+
+public interface ProjectDirectoryProvider {
```

Comment:
Can we inject a `ProjectLayout` instead of introducing a new service?

Reply:
Yes, thank you.  I really didn't want to add a new service, but somehow I got it in my head that `ProjectLayout` was not injectable and couldn't find another way to get the project dir.  Fixed.